extension/intents/tabs/tabs.js

```diff
@@ -104,6 +104,18 @@ intentRunner.registerIntent({
   },
 });
 
+intentRunner.registerIntent({
+  name: "tabs.closeAll",
+  async run(context) {
+    while (1) {
+      const activeTab = await context.activeTab();
+      if (activeTab.id === 1) break;
```

Comment:
I don't understand quite what's happening here? It seems like it's going to close the active tab until it has that particular id. But the id should be stable, and there's no reason there will necessarily be an id of 1. Like if you open the browser, open a few tabs. Close a few of those first tabs, then there will be no tab with id 1.

Also this will act on tabs in other windows.

Since closing _all_ tabs doesn't work well, we should either open a new tab or else leave the active tab open, or close all but pinned tabs. 

So a query that matches all tabs that should be closed might be: `browser.tabs.query({active: false, lastFocusedWindow: true, pinned: false})`

Reply:
@ianb I know that the id is not stable but for the first that opens up when you open up the browser always has id 1.
no matter how many times you open other tabs and close them. you can check that by using "console.log(activeTab.id);" . so my idea is to keep open that first tab with id 1.

Reply:
browser.tabs.query({active: false, lastFocusedWindow: true, pinned: false})  I tried this query in closeAll intent 
like : const openTabs = browser.tabs.query({active: false, lastFocusedWindow: true, pinned: false})
but it doesn't work. The intent runs but doesn't  close all tabs.

Reply:
@ianb  I have fixed it as you suggested.